src/main.rs

```diff
@@ -551,6 +825,48 @@ fn main() {
                 }
             }
 
+            if  device_config.is_audio_device().unwrap_or_default(){
+                
+                if taking_test_recoding && (Instant::now() - last_state_read).as_millis() > 1000{
+                        if let Ok(state) = read_tc2_agent_state(&mut dbus_conn){
+                             RP2040_STATE.store(state,Ordering::Relaxed);
+                             last_state_read = Instant::now();
+                            if state  &(0x04 | 0x08) == 0{
+                                taking_test_recoding = false;
+                            }
+                        }else{
+                            warn!("error reading tc2 agent state");
+                        }
+                    
+                }
+                else if TAKE_TEST_AUDIO.load(Ordering::Relaxed)
+                {
+                    if let Ok(state) = set_attiny_tc2_agent_test_audio_rec( &mut dbus_conn) {
+                        if safe_to_restart_rp2040(&mut dbus_conn) {
+                            // NOTE: Always reset rp2040 on startup if it's safe to do so.
+                            let _ = restart_tx.send(true);
+                            taking_test_recoding = true;
+                            info!("Telling rp2040 to take test recording and restarting");
+                        } 
+                        RP2040_STATE.store(state,Ordering::Relaxed);
+                        last_state_read = Instant::now();
+                    }
+                    TAKE_TEST_AUDIO.store(false,Ordering::Relaxed);
+                }
+                if !frame_acquire{
```

Comment:
Can this ever be true inside `if  device_config.is_audio_device().unwrap_or_default(){..}` ?